citesphere/src/main/java/edu/asu/diging/citesphere/web/user/UpdateItemsPageController.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroHttpStatusException;
+import edu.asu.diging.citesphere.core.service.ICitationCollectionManager;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.core.service.IGroupManager;
+import edu.asu.diging.citesphere.model.bib.impl.CitationResults;
+import edu.asu.diging.citesphere.user.IUser;
+
+@Controller
+@PropertySource("classpath:/config.properties")
+@PropertySource("classpath:/item_type_icons.properties")
+public class UpdateItemsPageController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Value("${_zotero_page_size}")
+    private Integer zoteroPageSize;
+
+    @Value("${_available_item_columns}")
+    private String availableColumns;
+
+    @Autowired
+    private Environment env;
+
+    @Autowired
+    private ICitationManager citationManager;
+
+    @Autowired
+    private ICitationCollectionManager collectionManager;
+
+    @Autowired
+    private IGroupManager groupManager;
+
+    @RequestMapping(value = { "/auth/group/{zoteroGroupId}/items/data",
+            "/auth/group/{zoteroGroupId}/collection/{collectionId}/items/data" })
+    public @ResponseBody ItemsDataDto show(Authentication authentication, @PathVariable("zoteroGroupId") String groupId,
+            @PathVariable(value = "collectionId", required = false) String collectionId,
+            @RequestParam(defaultValue = "1", required = false, value = "page") String page,
+            @RequestParam(defaultValue = "title", required = false, value = "sort") String sort,
+            @RequestParam(required = false, value = "columns") String[] columns) {
+        Integer pageInt = 1;
+        try {
+            pageInt = new Integer(page);
+        } catch (NumberFormatException ex) {
+            logger.warn("Trying to access invalid page number: " + page);
+        }
+        IUser user = (IUser) authentication.getPrincipal();
+        CitationResults results;
+        try {
+            results = citationManager.getGroupItems(user, groupId, collectionId, pageInt, sort);
+        } catch (ZoteroHttpStatusException e) {
+            logger.error("Zotero status exception occured while updating items page", e);
+            return null;
+        } catch (GroupDoesNotExistException e) {
+            logger.error("Group does not exist exception occured while updating items page", e);
+            return null;
```

Comment:
You want to make sure a 404 is returned in this case. This will probably return a 500.